### ISSUE TITLE
Great returns, but negative returns during the walkforward analysis

### ISSUE THREAD

**Issue by mathias5986** [freqtrade/freqtrade#3994]
Hello,

I have optimized a strategy by using the SharpeHyperOptLoss where I got a sharpe of 26 (-26 inverse) over 90 days (June 2020 - Aug 2020).
If I backtest with the parameters and the same period, I get the same great result:
<img width="698" alt="Screenshot 2020-11-24 at 23 43 40" src="https://user-images.githubusercontent.com/74938818/100160765-7ced9980-2eb0-11eb-89ea-88184466ecb3.png">

All good..

But if I do a walkforward analysis from Oct 2020-today, the results are extremely negative:
<img width="717" alt="Screenshot 2020-11-24 at 23 41 32" src="https://user-images.githubusercontent.com/74938818/100160955-d5249b80-2eb0-11eb-8873-6bfe33a0173a.png">

Was I just lucky the first time ? 
What should I do to get at least a positive return too ? Cancelling the optimization earlier with a lower sharpe ? if yes, what is the best realistic sharpe? With which sharpe are you happy?
Are there some tricks to avoid this? Maybe I need to backtest on 3 years and walkforward on 1 year?

Thanks!

**Comment by xmatthias** [comment]
markets change all the time - september / october were highly negative markets (especially september) - while june-august were kinda flat with a few sudden raises (which then fell again in september).

When picking a period to backtest, always make sure that it covers all market cycles, otherwise this will happen frequently.
Also doing that will not fully protect you though - as there will always be unseen events, that you can't really backtest.


in summary - not all strategies work great all the time, and markets change.
